tests/integration_tests/test_multi_2_sec.py

```diff
@@ -0,0 +1,409 @@
+        # aerodynamic lifting surface
+        multi_geom_group = MultiSecGeometry(
+            surface=surface, joining_comp=True, dim_constr=[np.array([1, 0, 0]), np.array([1, 0, 0])]
+        )
+        prob.model.add_subsystem(surface["name"], multi_geom_group)
+
+        # Generate the sections and unified mesh here in addition to adding the components.
+        # This has to ALSO be done here since AeroPoint has to know the unified mesh size.
+        section_surfaces = build_sections(surface)
+        uni_mesh = unify_mesh(section_surfaces)
+        surface["mesh"] = uni_mesh
+
+        # Create the aero point group, which contains the actual aerodynamic
+        # analyses
+        aero_group = AeroPoint(surfaces=[surface])
+        point_name = "aero_point_0"
+        prob.model.add_subsystem(
+            point_name, aero_group, promotes_inputs=["v", "alpha", "Mach_number", "re", "rho", "cg"]
+        )
+
+        # Get name of surface and construct unified mesh name
+        name = surface["name"]
+        unification_name = "{}_unification".format(surface["name"])
+
+        # Connect the mesh from the mesh unification component to the analysis point
+        prob.model.connect(
+            name + "." + unification_name + "." + name + "_uni_mesh", point_name + "." + "surface" + ".def_mesh"
+        )
+
+        # Perform the connections with the modified names within the
+        # 'aero_states' group.
+        prob.model.connect(
+            name + "." + unification_name + "." + name + "_uni_mesh",
+            point_name + ".aero_states." + "surface" + "_def_mesh",
+        )
+
+        prob.model.add_design_var("surface.sec0.chord_cp", lower=0.1, upper=10.0, units=None)
+        prob.model.add_design_var("surface.sec1.chord_cp", lower=0.1, upper=10.0, units=None)
+        prob.model.add_design_var("alpha", lower=0.0, upper=10.0, units="deg")
+
+        # Add joined mesh constraint
+        prob.model.add_constraint("surface.surface_joining.section_separation", upper=0, lower=0)
+        # prob.model.add_constraint('surface.surface_joining.section_separation',equals=0.0,scaler=1e-4)
+
+        # Add CL constraint
+        prob.model.add_constraint(point_name + ".CL", equals=0.3)
+
+        # Add Area constraint
+        prob.model.add_constraint(point_name + ".total_perf.S_ref_total", equals=2.0)
+
+        # Add objective
+        prob.model.add_objective(point_name + ".CD", scaler=1e4)
+
+        prob.driver = om.ScipyOptimizeDriver()
+        prob.driver.options["optimizer"] = "SLSQP"
+        prob.driver.options["tol"] = 1e-3
+        prob.driver.options["disp"] = True
+        prob.driver.options["maxiter"] = 1000
+        # prob.driver.options["debug_print"] = ["nl_cons", "objs", "desvars"]
+
+        # Set up and run the optimization problem
+        prob.setup()
+        prob.run_driver()
+
+        assert_near_equal(prob["aero_point_0.surface_perf.CD"][0], 0.02849356, 1e-6)
```

Comment:
I overlooked in the first review, but do you know why you get different results between the asym and sym cases? The surface dicts look like the same between two, so in theory, we should get the same results?

Is this because the b-spline works differently between sym and asym? If I understand it right, for a single-section surface, the root chord length = b-spline cp value at root for symmetry=True (because that's the anchor point), but for assymetric wing it's not always the case because the root chord length is not the anchor point but rather the "interpolated" value from b-spline. I don't know how this extends to the multi-section surface.

Reply:
The asymmetrical and symmetrical 2 section and 3 sections cases are fundamentally different problems.  For example an asymmetrical 3 section case only features 3 section along the entire span while the symmetrical 3 section case mean that there there are 3 sections in the half span and therefore 6 sections total. These a fundamentally different problems and give different results.